apps/src/lib/node/ledger/shell/mod.rs

```diff
@@ -854,35 +894,45 @@ mod test_utils {
     }
 
     impl TestShell {
-        /// Returns a new shell with
-        ///    - A broadcast receiver, which will receive any protocol txs sent
-        ///      by the shell.
-        ///    - A sender that can send Ethereum events into the ledger, mocking
-        ///      the Ethereum fullnode process
+        /// Returns a new test shell, and a number of channels to mock
+        /// shell sub-processes.
+        ///
+        /// The returned channels include:
+        ///
+        ///    - A channel which will receive any protocol txs sent by the shell
+        ///      via the [`Shell::broadcast`] method.
+        ///    - A channel that can send Ethereum events into the ledger,
+        ///      mocking the Ethereum fullnode process.
         pub fn new_at_height<H: Into<BlockHeight>>(
```

Comment:
I think we need to pass out the events receiver as well. 1. that way we can test it and 2. if it is dropped, finalize will fail when pushing events to it.

Reply:
ah yeah for sure, this is still a TODO item for another task (implementing the event log). I'll update the docstring and return type after that's implemented, this is pretty much just a PR in the spirit of "let's add this thing to the ledger, but we won't actually introduce any new functionality yet"